_posts/2018-11-13-type-constraints-correctness-reuse.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: Type constraints for code reuse and readability
+author: jguitana
+---
+
+The case for statically typed languages is so that we can prove our programs are right at compile time. Although our languages are not powerful enough to achieve this all the time, we certainly not always use their full capabilities. Our intuition should guide us to reduce implementation space and improve code reuse, allowing us to focus on small parts of a big problem. We can only do this if we choose the right abstraction in the form of Types, because these act as constraints on the implementation space, and if built correctly bring with them reusable and tested code.
+
+Let's take for example the `Int` and the `Nat` types. Int represents a number from `-x to x`. Nat represents a number from `0 to x`.
```

Comment:
I'd keep using \`Int\`  and  \`Nat\`.

Reply:
I thought about this too. But they are used quite often and everything becomes highlighted. I think it is better to highlight only when introducing for the first time. Does this make sense?

Reply:
Yes, that's why I didn't complain about this any further in the post.